Old Core API altered or broken?

From a clean workspace:
"sudo make install"
"tools/run_tests/build_python.sh"
"python2.7_virtual_environment/bin/python2.7 -B -m _adapter._low_test"

All of the EchoTest methods will fail with "AssertionError: Status(code=<Code.OK: 0>, details='') != Status(code=<Code.UNKNOWN: 2>, details='')".

Earlier in the evening I saw an GRPC_OP_COMPLETE emitted from the completion queue but I can't seem to reproduce that (it was in _adapter._blocking_invocation_inline_service_test, which now sees stream-response methods failing as timed out).

--- a/src/core/surface/call.c
+++ b/src/core/surface/call.c
@@ -404,7 +404,11 @@ static void get_final_status(grpc_call *call, grpc_ioreq_data out) {
       return;
     }
   }
-  out.recv_status.set_value(GRPC_STATUS_UNKNOWN, out.recv_status.user_data);
+  if (call->is_client) {
+    out.recv_status.set_value(GRPC_STATUS_UNKNOWN, out.recv_status.user_data);
+  } else {
+    out.recv_status.set_value(GRPC_STATUS_OK, out.recv_status.user_data);
+  }
 }
 
 static void get_final_details(grpc_call *call, grpc_ioreq_data out) {

